### PR TITLE
Move tests down from editor features to features layer.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/Microsoft.CodeAnalysis.CSharp.EditorFeatures.UnitTests.csproj [hunk 1]
```diff
@@ -69,7 +69,12 @@
     <EmbeddedResource Include="Debugging\Resources\*.*" LogicalName="Debugging/%(FileName)%(Extension)" />
   </ItemGroup>
   <ItemGroup>
-    <Compile Include="..\..\Analyzers\CSharp\Tests\UseVarTestExtensions.cs" Link="CodeRefactorings\UseVarTestExtensions.cs" />
+    <Compile Include="..\..\Features\CSharpTest\CodeActions\AbstractCSharpCodeActionTest.cs" Link="CodeActions\AbstractCSharpCodeActionTest.cs" />
```
Comment (dibarbet) [inline src/EditorFeatures/CSharpTest/Microsoft.CodeAnalysis.CSharp.EditorFeatures.UnitTests.csproj:72]: do the rest of the tests implementing `AbstractCSharpCodeActionTest` in editorfeatures make use of actual editor packages?  Or are you planning on moving those down later as well (similar q for the other linked files)

**Review by dibarbet — APPROVED** [review]
Only really looked in-depth on the actual modified files

**Comment by RikkiGibson** [comment]
Is this expected to mitigate some of the behavior seen in #67864?

**Comment by CyrusNajmabadi** [comment]
@RikkiGibson it might help there. This is mostly to help with the long test discover times I get locally :-)

**Pull request merged**